avahi-core/reconfirm.c

```diff
@@ -111,5 +113,30 @@ int avahi_record_reconfirm(
         return 0;
     }
 
+    /* If reconfirming an SRV record then also reconfirm the associated PTR record */
+    if (r->key->type == AVAHI_DNS_TYPE_SRV) {
```

Comment:
For my application it is very useful that the PTR record is automatically removed. Even if I would browse the list with my application's service type manually from my application, I would not know which entries should be reconfirmed (as the record's data with the PTR destination is not exposed in clear text via D-Bus). But the devices on my network also register a service subtype, which is not covered by this.

Example:
- Host: myhost.local
- Type: _http._tcp
- Subtype: _mysub._sub._http._tcp

So after reconfirmation has removed the SRV entry, the second PTR record (linking the subtype to my service) still persists.

myservice._http_tcp.local  IN      SRV 0 0 80 myhost.local  <-- removed by reconfirmation
_http._tcp.local   IN      PTR myservice._http._tcp.local  <-- removed by reconfirmation
_mysub._sub._http._tcp.local IN      PTR myservice._http._tcp.local <-- NOT covered by reconfirmation

Reply:
> as the record's data with the PTR destination is not exposed in clear text via D-Bus

It's exposed via D-Bus when clients subscribe to records with `RecordBrowser*`.

FWIW I'm not sure this API should automatically reconfirm RRs unless it's requested explicitly. There should probably be a high-level API that could purge all the RRs related to services.

Either way it's a good point and it would probably make sense to figure out how this API can be used.

Reply:
The API is meant to reconfirm individual resource records. Some RRs are tied to services, others to devices. I agree with what @evverx mentioned about not doing this automatically. There could be some high-level API to handle this, but for now applications that take that on themselves. Generally-speaking, though, I think the idea behind DNS-SD is that you are working with _services_, not _devices_.